### PR TITLE
Accept to param as function for redirect rule

### LINKED ISSUE #2: Accept function for the to parameter
Currently, the `to` parameter of a redirect rule can only be a string (optionally with regex groups). To give the developers even more freedom, why not accepting a function that will be executed and returns the correct `to` route based on the `from` route.

Example:

```js
{ from: '^/test/a*$', to: fromRoute => fromRoute.includes('aha') ? '/aha' : '/noo' }
```

### PR DESCRIPTION
**Pull request by juno-yu**
A preminilary impl which resolves #2 

(Sorry got a bug, async WIP)

### CODE REVIEW

**Comment by manniL** [comment]
Hey @juno-yu! Thanks for the PR and well done! I've added a small annotation with regards to `async/await` ☺️

File: lib/middleware.js [hunk 1]
```diff
@@ -9,6 +9,11 @@ function create (rules) {
     if (!foundRule) {
       return next()
     }
+    // Expect rule 'to' to be 1)regex or 2)string or 3)synchronous function taking from to compute to
+    let toTarget = foundRule.to
+    if (typeof toTarget === 'function') {
+      toTarget = toTarget(foundRule.from)
```
Comment (manniL) [inline lib/middleware.js]: It'd be great if the function could be `await`ed for in case it's async!

File: lib/middleware.js [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
+  return async function redirectRoute (req, res, next) {
     const decodedBaseUrl = decodeURI(req.url)
     const foundRule = rules.find(o => o.from.test(decodedBaseUrl))
 
     if (!foundRule) {
       return next()
     }
-    // Expect rule 'to' to be 1)regex or 2)string or 3)synchronous function taking from to compute to
+    // Expect rule 'to' to be 1)regex or 2)string or
+    // 3)function taking from & req (when from is regex, req might be more interesting)
```
Comment (juno-yu) [inline lib/middleware.js]: refer to the new test case

Comment (manniL) [reply lib/middleware.js]: Giving in the `req` as 2nd argument is smart! Like that ☺️

**Comment by manniL** [comment]
Alright, almost good to go! Could you update the `README` as well? 🙊

**Comment by manniL** [comment]
@juno-yu ping ☺️

**Pull request merged**